File: lib/extension.dart
import 'dart:js_interop';
import 'dart:js_interop_unsafe';

void main() {
  print('In Dart main!');
  globalContext.setProperty(
    'vscode_dart'.toJS,
    // Why doesn't any of this work, but putting something
    // like '1'.toJs does?
    // createJSInteropWrapper(Extension()),
    // Extension().activate,
    '1'.toJS,
  );
}

void activate() {
  print('Activated!');
}

@JSExport()
class Extension {
  @JSExport()
  void activate() {
    print('Activated');
  }

  @JSExport()
  void deactivate() {
    print('Deactivated');
  }
}

File: extension.js
const vscode = require('vscode');

function dartMainRunner() {
	console.log('dartMainRunner');
}

function activate(context) {
	globalThis.self = {};
	console.log('activate');
	const dartExtension = require('./out/extension.js');
	console.log(dartExtension);
	console.log(Object.keys(dartExtension));
	console.log(globalThis.self['vscode_dart']);
	console.log(globalThis.self['vscode_dart'].activate);
}

function deactivate() { }

module.exports = {
	activate,
	deactivate
}
